test/e2e/tests/vault-corruption/vault-corruption.spec.ts

```diff
@@ -0,0 +1,340 @@
+  }
+
+  /**
+   * Since reloading the background restarts the extension the UI isn't
+   * available immediately. So we just keep reloading the UI until it is. This
+   * is a bit of a hack, but I can't figure out a better way.
+   *
+   * @param driver - The WebDriver instance.
+   */
+  async function waitForVaultRestorePage(driver: Driver) {
+    let title: string;
+    do {
+      await driver.navigate(PAGES.HOME, {
+        waitForControllers: false,
+      });
+      title = await driver.driver.getTitle();
+      // the browser will return an error message for our UI's HOME page until
+      // the extension has restarted
+    } while (title !== WINDOW_TITLES.ExtensionInFullScreenView);
+  }
+
+  /**
+   * Breaks the databases and then begins recovery. Only returns once the
+   * background page has reloaded and the UI is available again.
+   *
+   * @param driver - The WebDriver instance.
+   * @param script - The script to break the DB that will be executed in the
+   * background page for MV2 or offscreen page for MV3.
+   * @returns The initial first account's address.
+   */
+  async function onboardThenCorruptVault(driver: Driver, script: string) {
+    const initialWindow = await driver.driver.getWindowHandle();
+
+    // open a spare tab so the browser doesn't exit once we `reload()` the
+    // extension process, as doing so will close all UI tabs we have open when
+    // we do -- and that will close the whole browser 😱
+    await driver.openNewPage('about:blank');
+
+    await onboard(driver);
+
+    const homePage = new HomePage(driver);
+    homePage.check_pageIsLoaded();
+
+    const headerNavbar = new HeaderNavbar(driver);
+    const firstAddress = await getFirstAddress(driver, headerNavbar);
+    await headerNavbar.lockMetaMask();
+
+    // use the home page to destroy the vault
+    await driver.executeAsyncScript(script);
+
+    // the previous tab we were using is now closed, so we need to tell Selenium
+    // to switch back to the other page (required for Chrome)
+    await driver.switchToWindow(initialWindow);
+
+    // get a new tab ready to use (required for Firefox)
+    await driver.openNewPage('about:blank');
+
+    // wait for the background page to reload
+    await waitForVaultRestorePage(driver);
+
+    return firstAddress;
+  }
+
+  /**
+   * Returns the vault recovery button.
+   *
+   * @param driver - The WebDriver instance.
+   */
+  async function getVaultRecoveryButton(driver: Driver) {
+    return await driver.waitForSelector('#critical-error-button', {
+      // The button is disabled for a timeout period as a precaution, so we
+      // need to wait for it to be enabled before continuing.
+      state: 'enabled',
+    });
+  }
+
+  /**
+   * Click the recovery/reset button then confirm or dismiss the action.
+   *
+   * @param driver - The WebDriver instance.
+   * @param confirm - Whether to confirm the action or not.
+   */
+  async function clickRecover(driver: Driver, confirm: boolean) {
```

Comment:
small suggestion: 
```suggestion
  async function clickRecover({driver: Driver, confirm: boolean}) {
```

if we used an object, it could slightly improve readability on the next steps, so instead of this:

`await clickRecover(driver, true);`

we could get this

`await clickRecover({ driver, confirm: true });`

wdyt?

Reply:
done!